apps/system/js/app_usage_metrics.js

```diff
@@ -805,8 +805,11 @@
     var addOn = false;
 
     if (app !== null && typeof app === 'object') {
+      // updateManifest is used by packaged apps until they are installed
+      var manifest = app.manifest ? app.manifest : app.updateManifest;
```

Comment:
If the manifest you get is the updateManifest, we won't have set the `addon` role yet, since we only know that once we have downloaded the package. So I'm not convinced that this will do what you need, but I haven't check when getAppUsage() is called.